Develop

IIExpanse opened the pull request:
Андрей, добрый день.
В проекте реализованы все задачи, кроме тех, которые выполняются после сдачи проекта.
Все ветки успешно прошли тесты постмана перед слиянием с develop.

IIExpanse committed:
feat: Реализовано удаление фильмов и пользователей по идентификатору.

IIExpanse committed:
feat: 1. Добавлен функционал по получению, добавлению, обновлению и удалению режиссеров. В класс Film добавлено поле с объектом режиссера.
2. Добавлено получение списка фильмов по режиссеру, отсортированное по дате выхода или лайкам.

IIExpanse committed:
fix: Доработка под требования тестов.

IIExpanse committed:
fix: Корректировка внутренних тестов.

IIExpanse committed:
Merge pull request #4 from Nagash40/add-director

Добавление режиссёров в фильмы

IIExpanse committed:
Merge remote-tracking branch 'origin/develop' into add-remove-endpoint

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/film/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/film/impl/InMemoryFilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/impl/InMemoryFilmStorage.java
#	src/main/resources/schema.sql

IIExpanse committed:
Merge pull request #5 from Nagash40/add-remove-endpoint

Add remove endpoint

Commit:
add functionality for film recommendation

Commit:
add integration test for getFilmRecommendation method

laserdi committed:
Добавление отзывов о фильмах. In Progress.
Создал таблицы с отзывами и лайками/дизлайками к ним.
Написана реализация интерфейса хранилища отзывов.
Осталось упорядочить структуру пакетов, проверить работу и написать контроллер.

Commit:
fix: update getFilmRecommendationTest()

Commit:
add more tests

Commit:
Merge remote-tracking branch 'origin/develop' into add-recommendations

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/dao/film/FilmDAO.java
#	src/main/java/ru/yandex/practicum/filmorate/service/film/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/film/impl/InMemoryFilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/impl/InMemoryFilmStorage.java
#	src/test/java/ru/yandex/practicum/filmorate/controller/UserControllerTest.java

Commit:
fix: compilation error to run application

Commit:
fix: update url path and remove films suggestion when there are no likes in order to handle all postman tests

Commit:
fix: update url path from 'recommendation' to 'recommendations'

Commit:
remove verification for getFilms()==null

laserdi committed:
Добавление отзывов о фильмах. In Progress.
Осталось всё доделать.

NoobProger-PT committed:
реализован поиск по названию и режиссеру

NoobProger-PT committed:
реализована сортировка популярных фильмов по жанру и году

NoobProger-PT committed:
реализована сортировка популярных фильмов по жанру и году

IIExpanse committed:
Merge pull request #7 from Nagash40/add-search

реализован поиск по названию и режиссеру

IIExpanse committed:
Merge pull request #8 from Nagash40/add-most-populars

реализована сортировка популярных фильмов по жанру и году

Commit:
Merge branch 'develop' into add-recommendations

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/film/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/film/impl/InMemoryFilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java

IIExpanse committed:
Merge pull request #6 from Nagash40/add-recommendations

Add recommendations

laserdi committed:
Добавление отзывов о фильмах. In Progress.
Осталось сделать вывод отзывов исходя из их рейтинга и необходимого количества.
Тесты PostMan ветки add-reviews: из 139 проходят 134.
В схему БД добавил стирание всех объектов из БД путём drop all objects.
В методе main печатается информация о запуске сервера поссле запуска программы.

IIExpanse committed:
fix: доработка под требования

IIExpanse committed:
fix: доработка под тесты

IIExpanse committed:
Merge remote-tracking branch 'origin/develop' into add-reviews

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/exception/GlobalExceptionHandler.java
#	src/main/resources/schema.sql

IIExpanse committed:
Merge pull request #9 from Nagash40/add-reviews

Add reviews

IIExpanse committed:
feat: добавлена лента событий

IIExpanse committed:
Merge pull request #10 from Nagash40/add-feed

feat: добавлена лента событий

IIExpanse committed:
feat: добавлена возможность получения общих фильмов по id пользователей

IIExpanse committed:
Merge pull request #11 from Nagash40/add-common-films

feat: добавлена возможность получения общих фильмов по id пользователей

IIExpanse committed:
feat: добавлена возможность получения общих фильмов по id пользователей

IIExpanse committed:
fix: удалены внутренние тесты, не соответствующие логике тестов постмана

shargin commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,60 @@
+
+    private final FilmService service;
+
+    public DirectorController(@Qualifier("FilmDBService") FilmService service) {
```
Тут и далее у интерфейса  FilmService присутствует только одна реализация - FilmDBService, квалификатор не требуется, Конструктор можно не объявлять явно, а использовать аннотацию ломбок

shargin commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,60 @@
+
+    @PostMapping
+    public ResponseEntity<Director> addDirector(@Valid @RequestBody Director director) {
+        int id = service.addDirector(director);
```
Чаще всего практичнее возвращать объект, который был сохранен. Из него пользователь может получить или id, если ему требуется, или использовать объект полностью. Например, так реализован репозиторий спринга https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save-S-. Например, тут бы его сразу можно было вернуть, и не создавать новый объект.
Review addReview работает именно так

Volody47 replied:
fixed

shargin commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,60 @@
+    @PutMapping
+    public ResponseEntity<Director> updateDirector(@Valid @RequestBody Director director) {
+        service.updateDirector(director);
+        log.debug("Обновлены данные о директоре с id={}", director.getId());
```
Лучше использовать единое именование (выше и ниже режиссер, а тут директор)

Volody47 replied:
fixed

shargin commented on src/main/java/ru/yandex/practicum/filmorate/exception/GlobalExceptionHandler.java:
```diff
@@ -27,7 +32,11 @@ public class GlobalExceptionHandler {
             UserNotFoundException.class,
             GenreNotFoundException.class,
-            MPANotFoundException.class
+            MPANotFoundException.class,
```
Желательно еще как-то обработать ошибки валидации ConstraintViolationException, например, отправляем некорректный запрос http :8080/films/popular?count=0? получаем 500 ошибку, а в стектрейсе видим getPopularFilms.count: должно быть больше 0, но пользователь не получил эту информацию, ее видит только владелец сервиса

shargin commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -36,8 +39,28 @@ public ResponseEntity<Collection<Film>> getFilmsList() {
+    @GetMapping("/search")
+    public ResponseEntity<Collection<Film>> searchFilms(@RequestParam String query,
+                                                        @RequestParam String by) {
+        return ResponseEntity.ok(service.searchFilms(query, by));
```
При получении фильмов режиссера используется enum для сортировки (что правильно), а тут нет

Commit:
fix: rename field директор to режиссерto be consistent; fix: update addDirector() to return Object

NoobProger-PT committed:
fix: в методе searchFilms используется enum

IIExpanse committed:
fix: 1. DAO приведены к единому стилю: методы создания и обновления объектов возвращают соответствующие объекты из базы.
2. Ошибки валидации перенесены в обработчик исключений, под все ошибки написаны собственные сообщения.
3. Обновлена схема БД.
4. Убраны ненужные квалификаторы. Ненужные конструкторы заменены аннотациями.
5. Опечатка "директор" исправлена на "режиссера".
6. Логика поиска переделана для использования enum.

IIExpanse committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/dao/film/FilmDAO.java
#	src/main/java/ru/yandex/practicum/filmorate/service/film/impl/FilmDBService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java

Nagash40 commented:
fix: 1. DAO приведены к единому стилю: методы создания и обновления объектов возвращают соответствующие объекты из базы.
2. Ошибки валидации перенесены в обработчик исключений, под все ошибки написаны собственные сообщения.
3. Обновлена диаграмма БД.
4. Убраны ненужные квалификаторы. Ненужные конструкторы заменены аннотациями.
5. Опечатка "директор" исправлена на "режиссера".
6. Логика поиска переделана для использования значений enum при определении типа поиска.

IIExpanse committed:
feat: Заполнен файл readme: указаны выполненные группой задачи.
